File: android/app/src/main/java/com/mart/distribution/demo/util/AreaNameUtil.kt
package com.mart.distribution.demo.util

private val AREA_NAME_ALIASES =
    mapOf(
        "warrangal" to "Warangal",
        "wrangal" to "Warangal",
        "warangal" to "Warangal",
        "warangal urban" to "Warangal Urban",
        "warangal rural" to "Warangal Rural",
    )

fun normalizeAreaName(raw: String): String {
    val trimmed = raw.trim().replace(Regex("\\s+"), " ")
    if (trimmed.isEmpty()) return trimmed
    return AREA_NAME_ALIASES[trimmed.lowercase()] ?: trimmed
}

File: backend/src/areas/area-name.util.ts
/** Canonical spellings for known territory names (case-insensitive keys). */
const AREA_NAME_ALIASES: Record<string, string> = {
  warrangal: 'Warangal',
  wrangal: 'Warangal',
  warangal: 'Warangal',
  'warangal urban': 'Warangal Urban',
  'warangal rural': 'Warangal Rural',
};

export function normalizeAreaName(raw: string): string {
  const trimmed = raw.trim().replace(/\s+/g, ' ');
  if (!trimmed) {
    return trimmed;
  }
  const alias = AREA_NAME_ALIASES[trimmed.toLowerCase()];
  return alias ?? trimmed;
}
